### PR TITLE
RE-1191 Control Log Storage v2

### PR DESCRIPTION
**Pull request by hughsaunders**
This PR adds two controls for log storage:

- Ensure that no console log may exceed 200MB - the job will be killed if that limit is exceeded. This is added via globalWraps, a new function that wraps every job, and will make it easier to add or remove global wrappers in future. This function allocates a shared slave, so shared_slave allocations have been removed from jobs where they are no longer necessary. Added a unit test for this functionality. 
- Ensure that all jobs have a retention policy so logs are not stored indefinitely. A lint rule is added to enforce this.

Issue: [RE-1191](https://rpc-openstack.atlassian.net/browse/RE-1191)

### CODE REVIEW

**Comment by hughsaunders** [comment]
This PR follows on from https://github.com/rcbops/rpc-gating/pull/577 which was reverted because one of the wrappers required a node to be allocated, but was called before any nodes were allocated.

**Comment by hughsaunders** [comment]
https://rpc.jenkins.cit.rackspace.net/job/RE-unit-test-globalwraps/2/ ✅  Unit test build succeeded, but that build will be deleted next time someone runs JJB.

**Comment by hughsaunders** [comment]
The functional and lint jobs failed due to a mismatch between the job, and branch being tested. The RE-1191 branch assumes that each job is wrapped in common.globalWraps so a node is allocated. However the current production jobs do not have this, so public cloud instance allocation fails as it needs a node to work from but doesn't have one. 

This mismatch cannot be solved without rearchitecting how we test PRs to rpc-gating. 

I build a test aio job from this branch and test it against this branch.

**Comment by hughsaunders** [comment]
RPC standard job built from this branch: https://rpc.jenkins.cit.rackspace.net/job/PM_rpc-openstack-newton-xenial_RE1191-swift-deploy/2/ ▶️

File: rpc_jobs/rpc_artifact_build.yml [hunk 1]
```diff
@@ -89,130 +89,130 @@
...
-        try {{
-          currentBuild.result = "SUCCESS"
-          // We need to checkout the rpc-openstack repo on the CIT Slave
-          // so that we can check whether the patch is a docs-only patch
-          // before allocating resources unnecessarily.
-          common.prepareRpcGit("auto", env.WORKSPACE)
-          if(common.is_doc_update_pr("${{env.WORKSPACE}}/rpc-openstack")){{
-            return
-          }}
+        common.globalWraps(){{
```
Comment (mattt416) [inline rpc_jobs/rpc_artifact_build.yml]: Indentation here is off.

**Comment by git-harry** [comment]
@hughsaunders what happened to re-adding https://github.com/rcbops/rpc-gating/pull/577/commits/2d3204b638835b7dce0accad165dfa157d1ea163? Is that being done separately?

File: rpc_jobs/rpc_artifact_build.yml (hunk 1 above)
Comment (hughsaunders) [reply rpc_jobs/rpc_artifact_build.yml]: Done

**Comment by hughsaunders** [comment]
@git-harry that must have disappeared either in an auto squash or a rebase, added back.

File: rpc_jobs/rpc_artifact_build.yml (hunk 1 above)
Comment (hughsaunders) [reply rpc_jobs/rpc_artifact_build.yml]: Done

Comment (mattt416) [reply rpc_jobs/rpc_artifact_build.yml]: DOUBLE DONE, just the way @major likes it.

**Comment by mattt416** [comment]
recheck_cit_unit_tests
recheck_xenial_functional_test
recheck_xenial_lint_test

**Comment by mattt416** [comment]
Let's try to merge this one in a controlled environment, so we're mindful if this introduces any issues or not.

**Comment by hughsaunders** [comment]
Rebased as there was a conflict with the unit tests job due to #610 merging.

**Comment by hughsaunders** [comment]
This is ready for merge,  the two CI failures are [expected](https://github.com/rcbops/rpc-gating/pull/602#issuecomment-358986830). @mattt416 what did you mean by controlled? early morning + monitoring?

**Comment by hughsaunders** [comment]
https://rpc.jenkins.cit.rackspace.net/job/PM_rpc-openstack-newton-xenial_RE1191-swift-deploy/1/ ✅ 
https://rpc.jenkins.cit.rackspace.net/job/RE-unit-test-globalwraps/1/ ✅

**Pull request merged**